theano/sandbox/cuda/dnn.py

```diff
@@ -368,6 +368,27 @@ def ensure_float(val, default, name):
     return val
 
 
+def do_merge_gpu_alloc_empty(node1, node2):
```

Comment:
I do not think we should put the check for op equility there. It is already done by MergeFeature before the first call to do_merge. But we call it again later before doing the swap. I suppose this test can be slow in some scan case. I don't want to slow down the optimization again. But I'll document that.